FreeRTOS: pico_flash/flash.c calls xTaskCreateAffinitySet which is not available using static memory allocation

pico_flash/flash.c uses xTaskCreateAffinitySet, which is only available in FreeRTOS if confiugred to support dynamic allocation.
https://github.com/raspberrypi/pico-sdk/blob/95ea6acad131124694cda1c162c52cd30e0aece0/src/rp2_common/pico_flash/flash.c#L148

It's defined here:
https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/974351fe4aec19779f5be774007810c6ab36a8ab/tasks.c#L1758

If one was trying to use pure static FreeRTOS
```
#define configSUPPORT_DYNAMIC_ALLOCATION    0
#define configKERNEL_PROVIDED_STATIC_MEMORY     1
```
you'll get the following compilation warning:
pico_flash/flash.c:148:23: warning: implicit declaration of function 'xTaskCreateAffinitySet'; did you mean 'vTaskCoreAffinitySet'? [-Wimplicit-function-declaration]

For static allocation, it looks like you'd first have to create the static task then use ```vTaskCoreAffinitySet```.

Perhaps it doesn't need to change (as pico sdk does other dynamic allocation, I believe), so maybe just outputting a warning during compilation  that says "pico sdk requires configSUPPORT_DYNAMIC_ALLOCATION to be 1" would be enough

--- a/src/rp2_common/pico_flash/flash.c
+++ b/src/rp2_common/pico_flash/flash.c
@@ -147,7 +147,7 @@ static int default_enter_safe_zone_timeout_ms(__unused uint32_t timeout_ms) {
         TaskHandle_t task_handle;
 
         // when FreeRTOS dynamic allocation is disabled (configSUPPORT_DYNAMIC_ALLOCATION == 0), the following instruction fails
-#ifndef configSUPPORT_DYNAMIC_ALLOCATION
+#if configSUPPORT_DYNAMIC_ALLOCATION
         if (pdPASS != xTaskCreateAffinitySet(flash_lockout_task, "flash lockout", configMINIMAL_STACK_SIZE, (void *)core_num, 0, 1u << (core_num ^ 1), &task_handle)) {
 #else
         static StackType_t flash_lockout_stack[configMINIMAL_STACK_SIZE];

--- a/src/rp2_common/pico_flash/flash.c
+++ b/src/rp2_common/pico_flash/flash.c
@@ -146,16 +146,14 @@ static int default_enter_safe_zone_timeout_ms(__unused uint32_t timeout_ms) {
         // create at low priority on other core
         TaskHandle_t task_handle;
 
-        // when FreeRTOS dynamic allocation is disabled (configSUPPORT_DYNAMIC_ALLOCATION == 0), the following instruction fails
+        // when FreeRTOS dynamic allocation is disabled (configSUPPORT_DYNAMIC_ALLOCATION == 0), the following fails
 #if configSUPPORT_DYNAMIC_ALLOCATION
         if (pdPASS != xTaskCreateAffinitySet(flash_lockout_task, "flash lockout", configMINIMAL_STACK_SIZE, (void *)core_num, 0, 1u << (core_num ^ 1), &task_handle)) {
 #else
         static StackType_t flash_lockout_stack[configMINIMAL_STACK_SIZE];
         static StaticTask_t flash_lockout_task_tcb;
-        task_handle = xTaskCreateStatic(flash_lockout_task, "flash lockout", configMINIMAL_STACK_SIZE, (void *)core_num, 0, flash_lockout_stack, &flash_lockout_task_tcb);
-        if (task_handle != NULL) {
-            vTaskCoreAffinitySet(task_handle, 1u << (core_num ^ 1));
-        } else {
+        task_handle = xTaskCreateStaticAffinitySet(flash_lockout_task, "flash lockout", configMINIMAL_STACK_SIZE, (void *)core_num, 0, flash_lockout_stack, &flash_lockout_task_tcb, 1u << (core_num ^ 1));
+        if (task_handle == NULL) {
 #endif
             return PICO_ERROR_INSUFFICIENT_RESOURCES;
         }

--- a/src/rp2_common/pico_flash/flash.c
+++ b/src/rp2_common/pico_flash/flash.c
@@ -145,7 +145,18 @@ static int default_enter_safe_zone_timeout_ms(__unused uint32_t timeout_ms) {
         uint core_num = get_core_num();
         // create at low priority on other core
         TaskHandle_t task_handle;
+
+        // when FreeRTOS dynamic allocation is disabled (configSUPPORT_DYNAMIC_ALLOCATION == 0), the following instruction fails
+#ifndef configSUPPORT_DYNAMIC_ALLOCATION
         if (pdPASS != xTaskCreateAffinitySet(flash_lockout_task, "flash lockout", configMINIMAL_STACK_SIZE, (void *)core_num, 0, 1u << (core_num ^ 1), &task_handle)) {
+#else
+        static StackType_t flash_lockout_stack[configMINIMAL_STACK_SIZE];
+        static StaticTask_t flash_lockout_task_tcb;
+        task_handle = xTaskCreateStatic(flash_lockout_task, "flash lockout", configMINIMAL_STACK_SIZE, (void *)core_num, 0, flash_lockout_stack, &flash_lockout_task_tcb);
+        if (task_handle != NULL) {
+            vTaskCoreAffinitySet(task_handle, 1u << (core_num ^ 1));
+        } else {
+#endif
             return PICO_ERROR_INSUFFICIENT_RESOURCES;
         }
         lockout_state[core_num] = FREERTOS_LOCKOUT_LOCKER_WAITING;

